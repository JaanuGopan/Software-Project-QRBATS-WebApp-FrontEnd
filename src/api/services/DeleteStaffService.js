import axios from "axios";
import ApiConstants from "../config/ApiConstants";

class DeleteStaffService {
  static async deleteStaff(staffId) {
    try {
      const response = await axios.post(ApiConstants.deleteStaffUrl, {
        userId: staffId,
      });
      console.log("deleted Successfully..");
      return response.data.token;
    } catch (error) {
      throw new Error("Delete Failed ", error);
    }
  }
}

export default DeleteStaffService;
