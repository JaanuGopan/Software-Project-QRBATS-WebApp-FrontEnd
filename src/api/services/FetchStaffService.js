import axios from "axios";
import ApiConstants from "../config/ApiConstants";

class FetchStaffService {
  static async fetchStaffs() {
    try {
      const response = await axios.post(ApiConstants.getAllStaffUrl);
      return response.data;
    } catch (error) {
      console.error("Error fetching staffs:", error);
      throw error; // Re-throw the error to handle it in the component
    }
  }
}

export default FetchStaffService;
