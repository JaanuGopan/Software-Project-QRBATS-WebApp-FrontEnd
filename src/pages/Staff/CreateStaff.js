import React, { useState } from "react";
import "./StaffA.css";
import axios from "axios";
import Designer from "../../assets/Images/Designer.jpeg";
import InputField from "../../components/textfields/InputBox/InputField";
import { IoMdCloseCircleOutline } from "react-icons/io";
import CreateUserService from "../../api/services/CreateUserService";

const CreateStaff = ({ handlecloseCreateStaffWindow }) => {
  const [firstname, setFirstName] = useState("");
  const [lastname, setLastName] = useState("");
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [confirmPassword, setConfirmPassword] = useState("");
  const [userName, setUserName] = useState("");
  const [userRole, setUserRole] = useState("");
  const [departmentId, setDepartmentId] = useState("");

  const deparmentList = ["DEIE", "DCOM", "DMME", "DCEE", "DMENA"];
  const userRoleList = ["ADMIN", "LECTURER"];

  const handleSubmit = async (e) => {
    e.preventDefault();
    try {
      const response = CreateUserService.saveUser(
        firstname,
        lastname,
        email,
        password,
        userName,
        deparmentList.indexOf(departmentId) + 1,
        userRoleList.indexOf(userRole)
      );

      console.error("Login success...");
      handlecloseCreateStaffWindow();
    } catch (error) {
      console.error("Login failed", error);
    }
  };

  return (
    <div className="staff-signup-main-container">
      <div
        className="closeCreateEventWindow"
        onClick={handlecloseCreateStaffWindow}
      >
        <IoMdCloseCircleOutline />
      </div>
      <p className="staff-head1">Create Staff</p>
      <div className="staff-login-container">
        <div className="staff-image-container">
          <img src={Designer} className="staff-logo" alt="Logo" />
        </div>
        <div className="form-container">
          <form onSubmit={handleSubmit}>
            <div className="choice-input mb-3">
              <select
                value={userRole}
                onChange={(e) => setUserRole(e.target.value)}
                className="student-select-input"
              >
                <option value="">Select user role</option>
                {userRoleList.map((option, index) => (
                  <option key={index} value={option}>
                    {option}
                  </option>
                ))}
              </select>
            </div>
            {/* First Name Input */}
            <InputField
              placeholder="Enter your first name"
              value={firstname}
              onChange={(e) => setFirstName(e.target.value)}
              inputType="text"
            />

            {/* Last Name Input */}
            <InputField
              placeholder="Enter your last name"
              value={lastname}
              onChange={(e) => setLastName(e.target.value)}
              inputType="text"
            />

            {/* Email Input */}
            <InputField
              placeholder="Enter your email"
              value={email}
              onChange={(e) => setEmail(e.target.value)}
              inputType="text"
            />

            <InputField
              placeholder="Enter your user name"
              value={userName}
              onChange={(e) => setUserName(e.target.value)}
              inputType="text"
            />

            {/* Password Input */}
            <InputField
              placeholder="Enter your password"
              value={password}
              onChange={(e) => setPassword(e.target.value)}
              inputType="password"
            />
            <InputField
              placeholder="Confirm password"
              value={confirmPassword}
              onChange={(e) => setConfirmPassword(e.target.value)}
              inputType="password"
            />

            <div className="choice-input mb-3">
              <select
                value={departmentId}
                onChange={(e) => setDepartmentId(e.target.value)}
                className="student-select-input"
              >
                <option value="">Select the department</option>
                {deparmentList.map((option, index) => (
                  <option key={index} value={option}>
                    {option}
                  </option>
                ))}
              </select>
            </div>
            <button type="submit" className="btn btn-primary w-100">
              Add Staff
            </button>
          </form>
        </div>
      </div>
    </div>
  );
};

export default CreateStaff;
