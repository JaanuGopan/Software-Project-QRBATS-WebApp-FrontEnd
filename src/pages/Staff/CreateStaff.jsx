import React, { useState } from 'react';
import './StaffA.css';
import InputField from '../../components/textfields/InputBox/InputField';
import { IoMdCloseCircleOutline } from 'react-icons/io';
import CreateUserService from '../../api/services/CreateUserService';
import Select from 'react-select';
import { toast } from 'react-toastify';
import { CircularProgress } from '@mui/material';
import Department from '../../utils/Department';

const CreateStaff = ({ handleCloseCreateStaffWindow, reloadStaffList }) => {
  const [firstName, setFirstName] = useState('');
  const [lastName, setLastName] = useState('');
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [confirmPassword, setConfirmPassword] = useState('');
  const [userName, setUserName] = useState('');
  const [userRole, setUserRole] = useState('');
  const [departmentId, setDepartmentId] = useState('');

  const departmentList = Department.departmentList;
  const userRoleList = ['ADMIN', 'LECTURER'];

  const notifySuccess = () => toast.success('Successfully Staff Created!');

  const [loadingCreateStaff, setLoadingCreateStaff] = useState(false);

  const handleInputValidation = () => {
    const emailPattern = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;

    if (firstName.trim() === '') {
      toast.error('Please enter first name');
      return false;
    }
    if (lastName.trim() === '') {
      toast.error('Please enter last name');
      return false;
    }
    if (email.trim() === '' || !emailPattern.test(email)) {
      toast.error('Please enter a valid email');
      return false;
    }
    if (userName.trim() === '') {
      toast.error('Please enter username');
      return false;
    }
    if (password.trim() === '') {
      toast.error('Please enter password');
      return false;
    }
    if (confirmPassword.trim() === '') {
      toast.error('Please confirm password');
      return false;
    }
    if (password !== confirmPassword) {
      toast.error('Passwords do not match');
      return false;
    }
    if (!departmentId) {
      toast.error('Please select department');
      return false;
    }
    if (!userRole) {
      toast.error('Please select user role');
      return false;
    }
    return true;
  };

  const handleSubmit = async (e) => {
    e.preventDefault();
    if (!handleInputValidation()) {
      return;
    }
    try {
      setLoadingCreateStaff(true);
      const response = await CreateUserService.saveUser(
        firstName,
        lastName,
        email,
        password,
        userName,
        departmentList.indexOf(departmentId.value) + 1,
        userRoleList.indexOf(userRole.value)
      );
      if (response.status === 200) {
        notifySuccess();
        reloadStaffList();
        handleCloseCreateStaffWindow();
        toast.success('User created successfully!');
      } else if (response.status === 400) {
        toast.error(response.data);
      }
    } catch (error) {
      console.error('User creation failed', error);
      toast.error('Error in user creation.');
    } finally {
      setLoadingCreateStaff(false);
    }
  };

  return (
    <div className="staff-signup-main-container">
      <div className="staff-update-title-close-button">
        <h3 className="staff-update-title">Create User</h3>
        <div className="staff-update-close-button" onClick={handleCloseCreateStaffWindow}>
          <IoMdCloseCircleOutline id="close-icon" />
        </div>
      </div>

      <div className="staff-login-container">
        <div className="form-container">
          <form onSubmit={handleSubmit}>
            <div className="staff-creation-input">
              <label>User Role</label>
              <div className="staff-creation-input-field">
                <Select
                  required
                  value={userRole}
                  onChange={(e) => setUserRole(e)}
                  options={userRoleList.map((role) => ({
                    value: role,
                    label: role,
                  }))}
                  placeholder={'Select User Role'}
                />
              </div>
            </div>
            <div className="staff-creation-input">
              <label>First Name</label>
              <div className="staff-creation-input-field">
                <InputField
                  placeholder="Enter your first name"
                  value={firstName}
                  onChange={(e) => setFirstName(e.target.value)}
                  inputType="text"
                />
              </div>
            </div>
            <div className="staff-creation-input">
              <label>Last Name</label>
              <div className="staff-creation-input-field">
                <InputField
                  placeholder="Enter your last name"
                  value={lastName}
                  onChange={(e) => setLastName(e.target.value)}
                  inputType="text"
                />
              </div>
            </div>

            <div className="staff-creation-input">
              <label>Email</label>
              <div className="staff-creation-input-field">
                <InputField
                  placeholder="Enter your email"
                  value={email}
                  onChange={(e) => setEmail(e.target.value)}
                  inputType="text"
                />
              </div>
            </div>
            <div className="staff-creation-input">
              <label>User Name</label>
              <div className="staff-creation-input-field">
                <InputField
                  placeholder="Enter your user name"
                  value={userName}
                  onChange={(e) => setUserName(e.target.value)}
                  inputType="text"
                />
              </div>
            </div>

            <div className="staff-creation-input">
              <label>Password</label>
              <div className="staff-creation-input-field">
                <InputField
                  placeholder="Enter your password"
                  value={password}
                  onChange={(e) => setPassword(e.target.value)}
                  inputType="password"
                />
              </div>
            </div>
            <div className="staff-creation-input">
              <label>Confirm Password</label>
              <div className="staff-creation-input-field">
                <InputField
                  placeholder="Confirm password"
                  value={confirmPassword}
                  onChange={(e) => setConfirmPassword(e.target.value)}
                  inputType="password"
                />
              </div>
            </div>
            <div className="staff-creation-input">
              <label>Department</label>
              <div className="staff-creation-input-field">
                <Select
                  required
                  value={departmentId}
                  onChange={(e) => setDepartmentId(e)}
                  options={departmentList.map((dept) => ({
                    value: dept,
                    label: dept,
                  }))}
                  placeholder={'Select Department'}
                />
              </div>
            </div>
            <div className="create-staff-create-button">
              {loadingCreateStaff ? (
                <div className="d-flex justify-content-center align-items-center">
                  <CircularProgress />
                </div>
              ) : (
                <button type="submit" className="btn btn-success">
                  Create Staff
                </button>
              )}
            </div>
          </form>
        </div>
      </div>
    </div>
  );
};

export default CreateStaff;
