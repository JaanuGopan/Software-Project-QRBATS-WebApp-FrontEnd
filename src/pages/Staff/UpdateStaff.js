import React, { useState } from "react";
import "./StaffA.css";
import axios from "axios";
import Designer from "../../assets/Images/Designer.jpeg";
import { useNavigate } from "react-router-dom";
import InputField from "../../components/textfields/InputBox/InputField";
import { IoMdCloseCircleOutline } from "react-icons/io";

const UpdateStaff = ({
  handlecloseUpdateStaffWindow,
  selectedStaff,
  handleReloadStaffList,
}) => {
  const [userId, setUserId] = useState(selectedStaff.userId);
  const [firstname, setFirstName] = useState(selectedStaff.firstName);
  const [lastname, setLastName] = useState(selectedStaff.lastName);
  const [email, setEmail] = useState(selectedStaff.email);
  const [password, setPassword] = useState("");
  const [userName, setUserName] = useState(selectedStaff.username);
  const [userRole, setUserRole] = useState(selectedStaff.role);
  const [departmentId, setDepartmentId] = useState(selectedStaff.departmentId);
  const navigate = useNavigate();

  const deparmentList = ["DEIE", "DCOM", "DMME", "DCEE", "DMENA"];
  const userRoleList = ["ADMIN", "LECTURER", "STAFF"];

  const handleSubmit = async (e) => {
    e.preventDefault();
    try {
      const response = await axios.post(
        "http://localhost:8080/api/v1/auth/signin",
        {
          userId: userId,
          firstName: firstname,
          lastName: lastname,
          email: email,
          password: password,
          userName: userName,
          departmentId: deparmentList.indexOf(departmentId) + 1,
        }
      );
      const token = response.data.token;
      const decodedToken = parseJwt(token);
      const userName = decodedToken.sub;
      localStorage.setItem("token", token);
      navigate("/mainNavigation", { state: { userName } });
    } catch (error) {
      console.error("Login failed", error);
    }
  };

  const parseJwt = (token) => {
    const base64Url = token.split(".")[1];
    const base64 = base64Url.replace(/-/g, "+").replace(/_/g, "/");
    const jsonPayload = decodeURIComponent(
      atob(base64)
        .split("")
        .map((c) => "%" + ("00" + c.charCodeAt(0).toString(16)).slice(-2))
        .join("")
    );
    return JSON.parse(jsonPayload);
  };

  return (
    <div className="staff-signup-main-container">
      <div
        className="closeCreateEventWindow"
        onClick={handlecloseUpdateStaffWindow}
      >
        <IoMdCloseCircleOutline />
      </div>
      <p className="staff-head1">Update Staff Details</p>
      <div className="staff-login-container">
        <div className="staff-image-container">
          <img src={Designer} className="staff-logo" alt="Logo" />
        </div>
        <div className="form-container">
          <form onSubmit={handleSubmit}>
            {/* First Name Input */}
            <InputField
              placeholder="Enter your first name"
              value={firstname}
              onChange={(e) => setFirstName(e.target.value)}
              inputType="text"
            />

            {/* Last Name Input */}
            <InputField
              placeholder="Enter your last name"
              value={lastname}
              onChange={(e) => setLastName(e.target.value)}
              inputType="text"
            />

            {/* Email Input */}
            <InputField
              placeholder="Enter your email"
              value={email}
              onChange={(e) => setEmail(e.target.value)}
              inputType="text"
            />

            <InputField
              placeholder="Enter your user name"
              value={userName}
              onChange={(e) => setUserName(e.target.value)}
              inputType="text"
            />

            {/* Password Input */}
            <InputField
              placeholder="Enter your password"
              value={password}
              onChange={(e) => setPassword(e.target.value)}
              inputType="password"
            />
            <div className="choice-input mb-3">
              <select
                value={departmentId}
                onChange={(e) => setDepartmentId(e.target.value)}
                className="staff-select-input"
              >
                <option value="">Select the department</option>
                {deparmentList.map((option, index) => (
                  <option key={index} value={option}>
                    {option}
                  </option>
                ))}
              </select>
            </div>
            <button type="submit" className="btn btn-primary w-100">
              Save
            </button>
          </form>
        </div>
      </div>
    </div>
  );
};

export default UpdateStaff;
