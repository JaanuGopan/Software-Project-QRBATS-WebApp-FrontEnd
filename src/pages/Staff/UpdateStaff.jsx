import React, { useState } from 'react';
import './StaffA.css';
import { useNavigate } from 'react-router-dom';
import InputField from '../../components/textfields/InputBox/InputField';
import { IoMdCloseCircleOutline } from 'react-icons/io';
import Select from 'react-select';
import UserService from '../../api/services/UserService';
import { toast, ToastContainer } from 'react-toastify';
import { CircularProgress } from '@mui/material';

const UpdateStaff = ({ handleCloseUpdateStaffWindow, selectedStaff, handleReloadStaffList }) => {
  const [userId, setUserId] = useState(selectedStaff.userId);
  const [firstName, setFirstName] = useState(selectedStaff.firstName);
  const [lastName, setLastName] = useState(selectedStaff.lastName);
  const [email, setEmail] = useState(selectedStaff.email);
  const [password, setPassword] = useState('');
  const [userName, setUserName] = useState(selectedStaff.username);
  const [userRole, setUserRole] = useState(selectedStaff.role);
  const [departmentId, setDepartmentId] = useState(selectedStaff.departmentId);
  const [department, setDepartment] = useState();
  const navigate = useNavigate();

  const departmentList = ['DEIE', 'DCOM', 'DMME', 'DCEE', 'DMENA', 'DIS'];
  const userRoleList = ['ADMIN', 'LECTURER'];

  const [loadingUpdateStaff, setLoadingUpdateStaff] = useState(false);

  const handleInputValidation = () => {
    const emailPattern = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;

    if (firstName.trim() === '') {
      toast.error('Please Enter First Name');
      return false;
    }
    if (lastName.trim() === '') {
      toast.error('Please Enter Last Name');
      return false;
    }
    if (email.trim() === '' || !emailPattern.test(email)) {
      toast.error('Please Enter a Valid Email');
      return false;
    }
    if (!departmentId) {
      toast.error('Please Select Department');
      return false;
    }
    if (userRole.trim() === '') {
      toast.error('Please Select User Role');
      return false;
    }
    return true;
  };

  const handleSaveUser = async (e) => {
    e.preventDefault();
    if (!handleInputValidation()) {
      return;
    }
    try {
      setLoadingUpdateStaff(true);
      const response = await UserService.updateUser(
        userId,
        firstName,
        lastName,
        email,
        '',
        '',
        departmentId
      );
      if (response.status === 200) {
        handleReloadStaffList();
        handleCloseUpdateStaffWindow();
        toast.success('User Updated Successfully!');
      } else if (response.status === 400) {
        toast.error(response.data);
      }
    } catch {
      toast.error('User Updation Failed!');
    } finally {
      setLoadingUpdateStaff(false);
    }
  };

  const parseJwt = (token) => {
    const base64Url = token.split('.')[1];
    const base64 = base64Url.replace(/-/g, '+').replace(/_/g, '/');
    const jsonPayload = decodeURIComponent(
      atob(base64)
        .split('')
        .map((c) => '%' + ('00' + c.charCodeAt(0).toString(16)).slice(-2))
        .join('')
    );
    return JSON.parse(jsonPayload);
  };

  return (
    <div className="staff-signup-main-container">
      <div className="staff-update-title-close-button">
        <h3 className="staff-update-title">Update User</h3>
        <div className="staff-update-close-button" onClick={handleCloseUpdateStaffWindow}>
          <IoMdCloseCircleOutline id="close-icon" />
        </div>
      </div>
      <div className="staff-login-container">
        <div className="form-container">
          <form onSubmit={handleSaveUser}>
            <div className="staff-creation-input">
              <label>First Name</label>
              <div className="staff-creation-input-field">
                <InputField
                  placeholder="Enter your first name"
                  value={firstName}
                  onChange={(e) => setFirstName(e.target.value)}
                  inputType="text"
                />
              </div>
            </div>
            <div className="staff-creation-input">
              <label>Last Name</label>
              <div className="staff-creation-input-field">
                <InputField
                  placeholder="Enter your last name"
                  value={lastName}
                  onChange={(e) => setLastName(e.target.value)}
                  inputType="text"
                />
              </div>
            </div>
            <div className="staff-creation-input">
              <label>Email</label>
              <div className="staff-creation-input-field">
                <InputField
                  placeholder="Enter your email"
                  value={email}
                  onChange={(e) => setEmail(e.target.value)}
                  inputType="text"
                />
              </div>
            </div>
            <div className="staff-creation-input">
              <label>Department</label>
              <div className="staff-creation-input-field">
                <Select
                  required
                  value={{
                    value: departmentList[departmentId - 1],
                    label: departmentList[departmentId - 1],
                  }}
                  onChange={(e) => setDepartmentId(departmentList.indexOf(e.value) + 1)}
                  options={departmentList.map((dept) => ({
                    value: dept,
                    label: dept,
                  }))}
                  placeholder={'Select Department'}
                />
              </div>
            </div>
            <div className="create-staff-create-button">
              {loadingUpdateStaff ? (
                <div className="d-flex justify-content-center align-items-center">
                  <CircularProgress />
                </div>
              ) : (
                <button type="submit" className="btn btn-warning">
                  <label>Update</label>
                </button>
              )}
            </div>
          </form>
        </div>
      </div>
    </div>
  );
};

export default UpdateStaff;
