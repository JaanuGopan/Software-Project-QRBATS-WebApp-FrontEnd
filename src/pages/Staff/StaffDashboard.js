import React, { useState, useEffect } from "react";
import "./StaffA.css";
import NormalButton from "../../components/layout/AdminDashboardComponent/NormalButton";
import { MdCreateNewFolder } from "react-icons/md";
import { RiDeleteBin5Fill } from "react-icons/ri";
import CreateStaff from "./CreateStaff";
import UpdateStaff from "./UpdateStaff";
import StaffTable from "../../components/layout/AdminDashboardComponent/StaffTable";
import FetchStaffService from "../../api/services/FetchStaffService";
import DeleteStaffService from "../../api/services/DeleteStaffService";

const StaffDashboard = () => {
  const [staffCreatePopUpWindow, setStaffCreatePopUpWindow] = useState(false);
  const [staffUpdatePopUpWindow, setStaffUpdatePopUpWindow] = useState(false);
  const [search, setSearch] = useState("");

  const [staffsList, setStaffsList] = useState([]);
  const [selectedStaff, setSelectedStaff] = useState(null);

  useEffect(() => {
    // Fetch the list of staffs from the API
    handleReloadStaffList();
  }, []);

  const handleEventClick = (staffs) => {
    setSelectedStaff(staffs);
    // Do whatever you want with the selected event data
    console.log("Selected Staffs:", staffs);
  };

  const handleDelete = async () => {
    try {
      const response = await DeleteStaffService.deleteStaff(
        selectedStaff.userId
      );
      handleReloadStaffList();
    } catch (error) {
      console.log("error " + error);
    }
  };

  const handleReloadStaffList = async () => {
    FetchStaffService.fetchStaffs()
      .then((staffs) => {
        setStaffsList(staffs);
      })
      .catch((error) => {
        console.error("Error fetching staffs:", error);
      });
  };

  return (
    <div className="staff-Dash">
      <div className="staff-SearchEvent">
        <p className="staff-mainHead">Staff Details</p>
        <input
          type="text"
          placeholder="Search..."
          style={{
            width: "150px",
            padding: "3px 40px",
            border: "0.5px solid black",
            borderRadius: "5px",
            textAlign: "center",
          }}
          onChange={(e) => setSearch(e.target.value)}
        />
      </div>
      <div className="staff-EventList">
        <StaffTable
          search={search}
          handleUpdateStaff={() => setStaffUpdatePopUpWindow(true)}
          onStaffClick={handleEventClick}
          staffsList={staffsList}
        />
        <div className="staff-List-Buttons">
          <NormalButton
            handleClick={() => setStaffCreatePopUpWindow(true)}
            title={"Create"}
            titlewithiconicon={
              <MdCreateNewFolder className="staff-buttonIcon" />
            }
          />
          <NormalButton
            title={"Delete"}
            handleClick={handleDelete}
            titlewithiconicon={
              <RiDeleteBin5Fill className="staff-buttonIcon" />
            }
          />
        </div>
      </div>
      {staffCreatePopUpWindow && (
        <div
          handleClick={() => setStaffCreatePopUpWindow(false)}
          className="staff-Create-Event-Dashboard"
        >
          <CreateStaff
            handlecloseCreateStaffWindow={() =>
              setStaffCreatePopUpWindow(false)
            }
          />
        </div>
      )}
      {staffUpdatePopUpWindow && (
        <div
          handleClick={() => setStaffCreatePopUpWindow(false)}
          className="staff-Create-Event-Dashboard"
        >
          <UpdateStaff
            handlecloseUpdateStaffWindow={() =>
              setStaffUpdatePopUpWindow(false)
            }
            selectedStaff={selectedStaff}
            handleReloadStaffList={handleReloadStaffList}
          />
        </div>
      )}
    </div>
  );
};

export default StaffDashboard;
