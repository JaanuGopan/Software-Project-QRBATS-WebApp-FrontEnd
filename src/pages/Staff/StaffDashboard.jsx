import React, { useState, useEffect } from 'react';
import './StaffA.css';
import NormalButton from '../../components/layout/AdminDashboardComponent/NormalButton';
import { MdCreateNewFolder } from 'react-icons/md';
import { RiDeleteBin5Fill } from 'react-icons/ri';
import CreateStaff from './CreateStaff';
import UpdateStaff from './UpdateStaff';
import StaffTable from '../../components/layout/AdminDashboardComponent/StaffTable';
import DeleteStaffService from '../../api/services/DeleteStaffService';
import { toast, ToastContainer } from 'react-toastify';
import WarningPopup from '../../components/warningPopup/WarningPopup';
import AdminService from '../../api/services/AdminService';

const StaffDashboard = () => {
  const [staffCreatePopUpWindow, setStaffCreatePopUpWindow] = useState(false);
  const [staffUpdatePopUpWindow, setStaffUpdatePopUpWindow] = useState(false);
  const [search, setSearch] = useState('');
  const [staffsList, setStaffsList] = useState([]);
  const [selectedStaff, setSelectedStaff] = useState(null);

  useEffect(() => {
    // Fetch the list of staffs from the API
    handleReloadStaffList();
  }, []);

  const handleEventClick = (staffs) => {
    setSelectedStaff(staffs);
    // Do whatever you want with the selected event data
    console.log('Selected Staffs:', staffs);
  };

  const [showDeletePopUpWindow, setShowDeletePopUpWindow] = useState(false);

  const [processingDeleteStaff, setProcessingDeleteStaff] = useState(false);

  const handleDelete = async () => {
    if (!selectedStaff) {
      toast.error('Please select a staff to delete');
      return;
    }
    try {
      setProcessingDeleteStaff(true);
      const response = await DeleteStaffService.deleteStaff(selectedStaff.userId);
      if (response.status === 200) {
        toast.success('Staff Deleted Successfully');
        setSelectedStaff(null);
        handleReloadStaffList();
      } else if (response.status === 400) {
        toast.error(response.data);
      }
    } catch (error) {
      console.log('error ' + error);
      toast.error('Error In Staff Deletion. ');
    } finally {
      setProcessingDeleteStaff(false);
      setShowDeletePopUpWindow(false);
    }
  };

  const handleReloadStaffList = async () => {
    AdminService.fetchStaffs()
      .then((staffs) => {
        setStaffsList(staffs);
      })
      .catch((error) => {
        console.error('Error fetching staffs:', error);
      });
  };

  return (
    <div className="staff-Dash">
      <ToastContainer />
      <div className="staff-SearchEvent">
        <p className="staff-mainHead">Staff Details</p>
        <input
          type="text"
          placeholder="Search..."
          style={{
            width: '150px',
            padding: '3px 40px',
            border: '0.5px solid black',
            borderRadius: '5px',
            textAlign: 'center',
          }}
          onChange={(e) => setSearch(e.target.value)}
        />
      </div>
      <div className="staff-EventList">
        <StaffTable
          search={search}
          handleUpdateStaff={() => setStaffUpdatePopUpWindow(true)}
          onStaffClick={handleEventClick}
          staffsList={staffsList}
        />
        <div className="staff-List-Buttons">
          <NormalButton
            handleClick={() => setStaffCreatePopUpWindow(true)}
            title={'Create'}
            titlewithiconicon={<MdCreateNewFolder className="staff-buttonIcon" />}
          />
          <NormalButton
            title={'Delete'}
            handleClick={() => {
              if (selectedStaff == null) {
                toast.error('Please Select Staff. ');
              } else {
                setShowDeletePopUpWindow(true);
              }
            }}
            titlewithiconicon={<RiDeleteBin5Fill className="staff-buttonIcon" />}
          />
        </div>
      </div>
      {staffCreatePopUpWindow && (
        <div className="staff-Create-Event-Dashboard">
          <CreateStaff
            handleCloseCreateStaffWindow={() => setStaffCreatePopUpWindow(false)}
            reloadStaffList={handleReloadStaffList}
          />
        </div>
      )}
      {staffUpdatePopUpWindow && (
        <div
          handleClick={() => setStaffCreatePopUpWindow(false)}
          className="staff-Create-Event-Dashboard"
        >
          <UpdateStaff
            handleCloseUpdateStaffWindow={() => setStaffUpdatePopUpWindow(false)}
            selectedStaff={selectedStaff}
            handleReloadStaffList={handleReloadStaffList}
          />
        </div>
      )}
      {showDeletePopUpWindow && selectedStaff && (
        <div className="student-creation-delete-popup-window">
          <WarningPopup
            handleOk={handleDelete}
            handleCloseWarningWindow={() => setShowDeletePopUpWindow(false)}
            buttonText={'Delete'}
            titleText={'Are you sure you want to delete this staff?'}
            processing={processingDeleteStaff}
          />
        </div>
      )}
    </div>
  );
};

export default StaffDashboard;
