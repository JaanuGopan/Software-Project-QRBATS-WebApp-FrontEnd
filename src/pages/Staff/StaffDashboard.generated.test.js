import renderer from 'react-test-renderer';
import React, { useState, useEffect } from "react";

import NormalButton from "../../components/layout/AdminDashboardComponent/NormalButton";
import { MdCreateNewFolder } from "react-icons/md";
import { RiDeleteBin5Fill } from "react-icons/ri";
import CreateStaff from "./CreateStaff";
import UpdateStaff from "./UpdateStaff";
import StaffTable from "../../components/layout/AdminDashboardComponent/StaffTable";
import FetchStaffService from "../../api/services/FetchStaffService";
import DeleteStaffService from "../../api/services/DeleteStaffService";
import StaffDashboard from "./StaffDashboard";

jest.mock("./StaffA.css");
jest.mock("../../components/layout/AdminDashboardComponent/NormalButton");
jest.mock("react-icons/md");
jest.mock("react-icons/ri");
jest.mock("./CreateStaff");
jest.mock("./UpdateStaff");
jest.mock("../../components/layout/AdminDashboardComponent/StaffTable");
jest.mock("../../api/services/FetchStaffService");
jest.mock("../../api/services/DeleteStaffService");

const renderTree = tree => renderer.create(tree);
describe('<StaffDashboard>', () => {
  it('should render component', () => {
    expect(renderTree(<StaffDashboard 
    />).toJSON()).toMatchSnapshot();
  });
  
});