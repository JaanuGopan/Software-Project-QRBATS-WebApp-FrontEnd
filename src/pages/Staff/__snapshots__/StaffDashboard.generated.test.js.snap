// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StaffDashboard> should render component 1`] = `
<div
  className="staff-Dash"
>
  <div
    className="staff-SearchEvent"
  >
    <p
      className="staff-mainHead"
    >
      Staff Details
    </p>
    <input
      onChange={[Function]}
      placeholder="Search..."
      style={
        Object {
          "border": "0.5px solid black",
          "borderRadius": "5px",
          "padding": "3px 40px",
          "textAlign": "center",
          "width": "150px",
        }
      }
      type="text"
    />
  </div>
  <div
    className="staff-EventList"
  >
    <div
      className="staff-List-Buttons"
    />
  </div>
</div>
`;
